lib/pure/os.nim

```diff
@@ -878,24 +883,27 @@ include "includes/oserr"
 when not defined(nimscript):
   include "includes/osenv"
 
-proc getHomeDir*(): string {.rtl, extern: "nos$1",
-  tags: [ReadEnvEffect, ReadIOEffect].} =
-  ## Returns the home directory of the current user.
-  ##
-  ## This proc is wrapped by the `expandTilde proc <#expandTilde,string>`_
-  ## for the convenience of processing paths coming from user configuration files.
-  ##
-  ## See also:
-  ## * `getConfigDir proc <#getConfigDir>`_
-  ## * `getTempDir proc <#getTempDir>`_
-  ## * `expandTilde proc <#expandTilde,string>`_
-  ## * `getCurrentDir proc <#getCurrentDir>`_
-  ## * `setCurrentDir proc <#setCurrentDir,string>`_
-  runnableExamples:
-    assert getHomeDir() == expandTilde("~")
+when not defined(nodejs):
+  proc getHomeDir*(): string {.rtl, extern: "nos$1",
+    tags: [ReadEnvEffect, ReadIOEffect].} =
+    ## Returns the home directory of the current user.
+    ##
+    ## This proc is wrapped by the `expandTilde proc <#expandTilde,string>`_
+    ## for the convenience of processing paths coming from user configuration files.
+    ##
+    ## See also:
+    ## * `getConfigDir proc <#getConfigDir>`_
+    ## * `getTempDir proc <#getTempDir>`_
+    ## * `expandTilde proc <#expandTilde,string>`_
+    ## * `getCurrentDir proc <#getCurrentDir>`_
+    ## * `setCurrentDir proc <#setCurrentDir,string>`_
+    runnableExamples:
+      assert getHomeDir() == expandTilde("~")
 
-  when defined(windows): return string(getEnv("USERPROFILE")) & "\\"
-  else: return string(getEnv("HOME")) & "/"
+    when defined(windows): return string(getEnv("USERPROFILE")) & "\\" 
+    else: return string(getEnv("HOME")) & "/"
+else:
+  proc getHomeDir*(): string {.inline.} = jsos.homedir()
```

Comment:
You can just implement this and everything else as an another `when` branch instead of doing it like this.